### ISSUE TITLE
Support batch ingest of non-Work records

### ISSUE THREAD

**Issue by mcritchlow** [ucsdlib/horton#48]
### Descriptive summary

DAMS Manager currently provides support for batch ingesting subjects. This feature needs to be ported to horton, with support for new linked record object types:

- Agent
- Concept
- Place

Currently assuming the file format for ingest is Excel `.xlsx`

### Rationale

The pre-ingesting of records that Works will link to is part of the current DAMS4 workflow that @ucsdlib/domm would like to continue and port to DAMS5.

This feature would be used (optionally) prior to the delimited batch ingest for Works.

### Related work

#46

**Comment by ghost** [comment]
Yes, we will continue to use Excel .xlsx for the ingest file.  Updated template to be provided.

New feature for Pre-ingest validation
- "Search existing DAMS records"
  - Purpose: This will run before import and is similar to the ARK report generated after import by the current subject import tool.  It will allow the user to know which terms already have DAMS ARKS so they only have to import new terms.  It will also allow the user to check that an existing term is really and exact match to their source metadata.
  - Output report columns:
1. Action
2. Type
3. Term
4. ARK
5. exactMatch  (if present)
6. Example DAMS Object ARK

**Comment by ghost** [comment]
Here is the sample import Excel file.
I added alternateLabel for cases where we are changing the label from the data provider to a different form - we still want access via that other form.
[Uploading non-object_import.xlsx…]()

**Comment by lsitu** [comment]
@GregReser It looks like something is wrong with the sample import Excel file "Uploading non-object_import.xlsx…" that you attached. Could you take a look?

**Comment by ghost** [comment]
@lsitu Can't get the spreadsheet to upload, so I sent it to you in an email

**Comment by lsitu** [comment]
Thanks @GregReser. Yes, I see we can't attach the XML format files in the past and I have to use a different file extensions like .txt. But it looks like it should support files types like PNG, GIF, JPG, DOCX, PPTX, XLSX, TXT, PDF, or ZIP.

**Comment by lsitu** [comment]
@GregReser I see the following Agent types in the sample import Excel file:
Agent:Person
Agent:Group
Agent:Organization

From the [data dictionary](https://docs.google.com/spreadsheets/d/1OjTm1Kuzo-An-THdpUkjwa5puul-b4Lx3SK0lB6rvl8/edit#gid=616154196), I am a little confusing with their complexities:

- Are these ucsd:Agent instances with types in FOAF (http://xmlns.com/foaf/spec/) for foaf:Person, foaf:Group, foaf:Organization respectively, which are sub-class of foaf:Agent? 
- Are these also edm:Agent types as they derive all the properties from edm:Agent, so as they are the instances of ucsd:Agent, edm:Agent, foaf:Agent, and foaf:Person|foaf:Group|foaf:Organization respectively?

**Comment by ghost** [comment]
@lsitu We have added several new columns to the template.  I have created a draft for our discussions at:
https://docs.google.com/a/ucsd.edu/spreadsheets/d/1EVfMBmWfamqEXZWE9mgWMZPGT_7zN8wGtYArZisJUXA/edit?usp=sharing

Since we are still discussing some issues, like ORCID, there may be more changes coming.  :)
